crates/paimon/src/catalog/mod.rs

```diff
@@ -0,0 +1,254 @@
+
+    /// Creates a new table.
+    async fn create_table(
+        &self,
+        identifier: &Identifier,
+        schema: TableSchema,
+        ignore_if_exists: bool,
+    ) -> Result<()>;
+
+    /// Renames a table.
+    async fn rename_table(
+        &self,
+        from_table: &Identifier,
+        to_table: &Identifier,
+        ignore_if_not_exists: bool,
+    ) -> Result<()>;
+
+    /// Alters an existing table.
+    async fn alter_table(
+        &self,
+        identifier: &Identifier,
+        changes: Vec<SchemaChange>,
+        ignore_if_not_exists: bool,
+    ) -> Result<()>;
+
+    /// Drops a partition from the specified table.
+    async fn drop_partition(
+        &self,
+        identifier: &Identifier,
+        partitions: &HashMap<String, String>,
+    ) -> Result<()>;
+
+    /// Returns whether this catalog is case-sensitive.
+    fn case_sensitive(&self) -> bool {
+        true
+    }
+}
+
+/// Identifies an object in a catalog.
+///
+/// Impl References: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/catalog/Identifier.java#L35>
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Identifier {
+    database: String,
+    table: String,
+}
+
+impl Identifier {
+    pub const UNKNOWN_DATABASE: &'static str = "unknown";
+
+    /// Create a new identifier.
+    pub fn new(database: String, table: String) -> Self {
+        Self { database, table }
+    }
+
+    /// Get the table name.
+    pub fn database_name(&self) -> &str {
+        &self.database
+    }
+
+    /// Get the table name.
+    pub fn object_name(&self) -> &str {
+        &self.table
+    }
+
+    /// Get the full name of the identifier.
+    pub fn full_name(&self) -> String {
+        if self.database == Self::UNKNOWN_DATABASE {
+            self.table.clone()
+        } else {
+            format!("{}.{}", self.database, self.table)
+        }
+    }
+
+    /// Get the full name of the identifier with a specified character.
+    pub fn escaped_full_name(&self) -> String {
+        self.escaped_full_name_with_char('`')
+    }
+
+    /// Get the full name of the identifier with a specified character.
+    pub fn escaped_full_name_with_char(&self, escape_char: char) -> String {
+        format!(
+            "{0}{1}{0}.{0}{2}{0}",
+            escape_char, self.database, self.table
+        )
+    }
+
+    /// Create a new identifier.
+    pub fn create(db: &str, table: &str) -> Self {
+        Self::new(db.to_string(), table.to_string())
+    }
+}
+
+impl fmt::Display for Identifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.full_name())
+    }
+}
+
+/// A table provides basic abstraction for a table type and table scan, and table read.
+///
+/// Impl Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/table/Table.java#L41>
+pub trait Table {
```

Comment:
I feel like `Table` should be a struct instead of a trait.

Reply:
I will create a simple `Table` struct as the return value of the get_table method. With more design details about `Table` we can create a new issue for discussion. What do you think of it?